geode-management/src/main/java/org/apache/geode/management/api/ClusterManagementOperationStatusResult.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.management.api;
+
+
+
+import org.apache.geode.annotations.Experimental;
+
+@Experimental
+public class ClusterManagementOperationStatusResult<V extends JsonSerializable>
```

Comment:
Now that you've used this to convey the completed result, I think we can combine the ClusterManagementOperationResult with the ClusterManagementOperationStatusResult.

Reply:
it's not obvious to me how you would combine them.  one is Future<V> and one is V.  We had them combined before and it necessitated the extra CompletedAsyncOperationResult with the hack to set it to null in one case.  We still have the same number of classes now, but less indirection and the null hack is gone.

Reply:
And this class can be internal but ClusterManagementOperationResult needs to be in the external api

Reply:
explored some options but decided there is no viable way to combine them without CompletableFuture internals getting inadvertently serialized as well